src/CPULSSolver.cpp

```diff
@@ -243,41 +243,26 @@ void CPULSSolver::computeFSRSources(int iteration) {
         FP_PRECISION* scatter_sources_x = _groupwise_scratch.at(tid);
         for (int g_prime=0; g_prime < _num_groups; g_prime++) {
           int idx = first_scattering_index + g_prime;
-          if (sigma_s[idx] > 0) {
-            scatter_sources_x[g_prime] = sigma_s[idx] *
-              _scalar_flux_xyz(r,g_prime,0);
-          }
-          else {
-            scatter_sources_x[g_prime] = 1.0e-20;
-          }
+          scatter_sources_x[g_prime] = sigma_s[idx] *
+               _scalar_flux_xyz(r,g_prime,0);
```

Comment:
There is an option called _stabilize_moments, which is always TRUE. It means the XYZ linear scalar flux moments are always corrected in the same stabilization way as the flat scalar flux. Therefore, I agree with your changes here. Is there any numerical change after you fix this bug?
However, if we set the _stabilize_moments to be FALSE, will it be necessary to keep the if (sigma_s[idx] > 0) clause to make the convergence more stable? It would be interesting if you do some test about this. @GiudGiud

Reply:
There's about 10 pcms on a 3D pincell.
The (sigma_s[idx] > 0) clause makes the moments wrong whether we have stabilization or not.
I get good convergence with transport corrected cross-sections without any stabilization of MOC for the 3D BEAVRs assembly right now.

Reply:
Could you try to re-run the examples mentioned in Geoff's thesis that were unstable without his stabilization?